apps/server/src/modules/learnroom/uc/rooms.uc.ts

```diff
@@ -0,0 +1,78 @@
+import { Injectable, NotFoundException, UnauthorizedException } from '@nestjs/common';
+import { EntityId, Course, Task, TaskWithStatusVo } from '@shared/domain';
+import { CourseRepo, TaskRepo } from '@shared/repo';
+
+// TODO: move this somewhere else
+export interface Board {
+	roomId: string;
+	displayColor: string;
+	title: string;
+	elements: BoardElement[];
+}
+
+export type BoardElement = {
+	// TODO: should become fullblown class
+	type: string;
+	content: TaskWithStatusVo;
+};
+
+@Injectable()
+export class RoomsUc {
+	constructor(private readonly courseRepo: CourseRepo, private readonly taskRepo: TaskRepo) {}
+
+	async getBoard(roomId: EntityId, userId: EntityId): Promise<Board> {
+		const course = await this.courseRepo.findOne(roomId, userId);
+		const courseRole = this.getRoleInCourse(userId, course);
+		const taskFilter = this.taskFilter(courseRole);
+		const [tasks] = await this.taskRepo.findBySingleParent(course.id, taskFilter);
+		const tasksWithStatusVos = this.addStatusToTasks(courseRole, tasks, userId);
+
+		const board = this.buildBoard(course, tasksWithStatusVos);
+		return board;
+	}
+
+	private taskFilter(courseRole: string): { draft?: boolean } {
+		const shouldShowDrafts = courseRole === 'teacher';
+		return { draft: shouldShowDrafts };
+	}
+
+	// TODO: move somewhere else
```

Comment:
still need to move this?

Reply:
yes, but into a class that doesnt exist yet. we will do that in a future sprint.